Sources/IO/Misc/PDBReader/index.js

```diff
@@ -0,0 +1,257 @@
+import * as macro from '../../../macro';
+import vtkPolyData from '../../../Common/DataModel/PolyData';
+import vtkDataArray from '../../../Common/Core/DataArray';
+import DynamicTypedArray from '../../../Common/Core/DynamicTypedArray';
+
+import DataAccessHelper from '../../Core/DataAccessHelper';
+
+import atomElem from '../../../../Utilities/XMLConverter/chemistry/elements.json';
+
+
+// ----
+// Globals
+// ----
+
+const ATOMS = {};
+atomElem.atoms.forEach((a) => { ATOMS[a.id] = a; });
+
+// ----------------------------------------------------------------------------
+// vtkPDBReader methods
+// ----------------------------------------------------------------------------
+
+export function vtkPDBReader(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkPDBReader');
+
+  // Create default dataAccessHelper if not available
+  if (!model.dataAccessHelper) {
+    model.dataAccessHelper = DataAccessHelper.get('http');
+  }
+
+  // Internal method to fetch Array
+  function fetchPDB(url) {
+    return model.dataAccessHelper.fetchText(publicAPI, url);
+  }
+
+  // Set DataSet url
+  publicAPI.setUrl = (url) => {
+    if (url.indexOf('.pdb') === -1) {
+      model.baseURL = url;
+      model.url = `${url}`; // `${url}/index.pdb`;
+    } else {
+      model.url = url;
+
+      // Remove the file in the URL
+      const path = url.split('/');
+      path.pop();
+      model.baseURL = path.join('/');
+    }
+
+    // Fetch metadata
+    return publicAPI.loadData();
+  };
+
+  // Fetch the actual data arrays
+  publicAPI.loadData = () => {
+    const promise = fetchPDB(model.url);
+
+    promise.then(
+      (pdb) => {
+        model.pdb = pdb;
+        model.molecule = [];
+
+        // Parse data
+        model.molecule = model.pdb.split('\n');
+
+        publicAPI.modified();
+      });
+
+    return promise;
+  };
+
+  publicAPI.requestData = (inData, outData) => {
+    const polydata = vtkPolyData.newInstance();
+
+    if (model.molecule) {
+      const jSize = model.molecule.length;
+
+      const points = polydata.getPoints();
+      const pointdata = polydata.getPointData();
+
+      // atom position
+      const pointValues = new DynamicTypedArray({ chunkSize: 65500, arrayType: 'Float32Array' }); // points.getData();
```

Comment:
We can know the size notwithstanding to do a loop in addition (Check the number of 'ATOMS' command and count all atoms which are not Hydrogen if we want to stick to the [PDBReader vtk version](https://github.com/Kitware/VTK/blob/master/IO/Geometry/vtkPDBReader.cxx#L91-L92))